android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java

```diff
@@ -0,0 +1,676 @@
+                                .setSyncType("CertificateSync")
+                                .setSyncProgress(Long.valueOf(1))
+                                .setErrorCode("").build();
+
+                        result.success(sync);
+                        return;
+                    } else
+                        sync = new SyncPigeon.Sync.Builder()
+                                .setSyncType("CertificateSync")
+                                .setSyncProgress(Long.valueOf(1))
+                                .setErrorCode("Policy key Sync failed").build();
+                    result.success(sync);
+                    return;
+                } else
+                    sync = new SyncPigeon.Sync.Builder()
+                            .setSyncType("CertificateSync")
+                            .setSyncProgress(Long.valueOf(1))
+                            .setErrorCode("Policy key Sync failed").build();
+                result.success(sync);
+                return;
+            }
+
+            @Override
+            public void onFailure(Call<ResponseWrapper<CertificateResponse>> call, Throwable t) {
+                sync = new SyncPigeon.Sync.Builder()
+                        .setSyncType("CertificateSync")
+                        .setSyncProgress(Long.valueOf(1))
+                        .setErrorCode("Policy key Sync failed").build();
+                result.success(sync);
+                return;
+            }
+        });
+    }
+
+    private void syncGlobalParamsData(@NonNull SyncPigeon.Result<SyncPigeon.Sync> result) throws Exception {
+        Log.i(TAG, "config data sync is started");
+        Call<ResponseWrapper<Map<String, Object>>> call = syncRestService.getGlobalConfigs(
+                clientCryptoManagerService.getClientKeyIndex(), BuildConfig.CLIENT_VERSION);
+        call.enqueue(new Callback<ResponseWrapper<Map<String, Object>>>() {
+            @Override
+            public void onResponse(Call<ResponseWrapper<Map<String, Object>>> call, Response<ResponseWrapper<Map<String, Object>>> response) {
+                if (response.isSuccessful()) {
+                    ServiceError error = SyncRestUtil.getServiceError(response.body());
+                    if (error == null) {
+                        saveGlobalParams(response.body().getResponse());
+                        sync = new SyncPigeon.Sync.Builder()
+                                .setSyncType("GlobalParamsSync")
+                                .setSyncProgress(Long.valueOf(2))
+                                .setErrorCode("")
+                                .build();
+                        result.success(sync);
+                        return;
+                    } else
+                        sync = new SyncPigeon.Sync.Builder()
+                                .setSyncType("GlobalParamsSync")
+                                .setSyncProgress(Long.valueOf(2))
+                                .setErrorCode("GlobalParamsSync Failed")
```

Comment:
This should be an errorCode not the errorMessage. Error code could be config_sync_failed and the message for this errorCode to be maintained in I10n files